src/util.js

```diff
@@ -107,3 +107,55 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+export function updateContainers() {
+    let newContainers = {};
+    let newContainersStats = {};
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                .then(reply => {
+                    let newContainersMeta = reply.containers;
+                    let inspectRet = newContainersMeta.map((container) => {
+                        return new Promise((resolve, reject) => {
+                            varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id})
+                                    .then(reply => resolve(JSON.parse(reply.container)))
+                                    .catch(ex => reject(new Error("Failed to do InspectContainer call:", ex, JSON.stringify(ex))));
+                        });
+                    });
+
+                    Promise.all(inspectRet)
+                            .then((inspectRet) => {
+                                inspectRet.map((inspectRetEle) => {
+                                    newContainers[inspectRetEle.ID] = inspectRetEle;
+                                });
+                                resolve({newContainers: newContainers, newContainersStats: newContainersStats});
```

Comment:
It's too early to resolve the top-level promise here, as you still need to process the ContainerStats() results. Just drop this.